src/dist/worker.cc

```diff
@@ -236,44 +345,93 @@ agd::Status Worker::Run(const Params& params, const Parameters& aligner_params,
       }
     }
 
-    cout << "Work queue thread ending.\n";
+    cout << "Result queue thread ending.\n";
   });
 
-  timestamps_.reserve(100000);
-  queue_sizes_.reserve(100000);
+  incomplete_request_queue_thread_ = thread([this, &params]() {
+    auto free_func = [](void* data, void* hint) {
+      delete reinterpret_cast<char*>(data);
+    };
+    while (run_ && !(irqt_signal_ && incomplete_request_queue_->size() == 0)) {
+      MarshalledRequest request;
 
-  queue_measure_thread_ = std::thread([this](){
-      // get timestamp, queue size
-      //cout << "queue measure thread starting ...\n";
-      while(run_) {
-        time_t result = std::time(nullptr);
-        timestamps_.push_back(static_cast<long int>(result));
-        queue_sizes_.push_back(work_queue_->size());
-        std::this_thread::sleep_for(std::chrono::milliseconds(500));
-        if (queue_sizes_.size() >= 1000000) {
-          break;  // dont run forever ... 
-        }
+      if (!incomplete_request_queue_->pop(request)) {
+        continue;
+      }
+
+      auto size = request.buf.size();
+      zmq::message_t msg(request.buf.release_raw(), size, free_func, NULL);
+      bool success = zmq_incomplete_request_socket_->send(std::move(msg));
+      if (!success) {
+        cout << "INCOMP REQ Thread failed to send response over zmq!\n";
       }
-      //cout << "queue measure thread finished\n";
     }
-  );
+
+    cout << "Incomplete request queue thread ending.\n";
+  });
+
+  timestamps_.reserve(100000);
+  queue_sizes_.reserve(100000);
+
+  // queue_measure_thread_ = std::thread([this](){
+  //     // get timestamp, queue size
+  //     //cout << "queue measure thread starting ...\n";
+  //     while(run_) {
+  //       time_t result = std::time(nullptr);
+  //       timestamps_.push_back(static_cast<long int>(result));
+  //       queue_sizes_.push_back(work_queue_->size());
+  //       std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  //       if (queue_sizes_.size() >= 1000000) {
+  //         break;  // dont run forever ...
+  //       }
+  //     }
+  //     cout << "queue measure thread finished\n";
+  //   }
+  // );
+
+  // Raises SIGUSR1 in 15 secs time
```

Comment:
if this stuff doesnt need to be here delete it.

Reply:
Okay!